runtime/lua/vim/diagnostic.lua

```diff
@@ -524,6 +542,9 @@ end
 ---                       severity |diagnostic-severity|
 ---                       * source: (string) Include the diagnostic source in virtual
 ---                       text. One of "always" or "if_many".
+---                       * format: (function) A function that takes a diagnostic as input and
```

Comment:
good practice to actually write the function signature, or show an example. Sometimes is less verbose too.

Reply:
example is usually ideal. But it might lower the friction of documentation if we find a convention for describing signatures (probably just whatever https://github.com/teal-language/tl does)